### ISSUE TITLE
integrate with react-router v4

### ISSUE THREAD

**Issue by burczu** [burczu/react-router-role-authorization#12]
As the new version of react-router is completely rewriten, it would be nice to add support for this version of react-router. Or maybe it is completely unnecesary now?

**Comment by clairekh** [comment]
Are you planning support react-router v4?

**Comment by mstruensee** [comment]
bump ... Are you planning support react-router v4?

**Comment by burczu** [comment]
yeah, there is such plan but I had no time for that recently... I will try to take a look at this in August

**Comment by mstruensee** [comment]
if it helps at all, i had it (or so i thought) implemented ... it just didn't get the "roles" from the route ... when i hard coded things, it hid and showed as expected ...

thanks for the update.

edit:
After doing some research, the first issue I noticed was this ...
You should not use <Route component> and <Route children> in the same route; <Route children> will be ignored

the proposed solution is to change 
`<Route authorize={['admin']} component={RestrictedContainer}>
        <Route component={RestrictedPageComponent} path="/restricted" />
</Route>`
to ... 
`<RestrictedContainer` authorize={['admin']}>
        <Route component={RestrictedPageComponent} path="/restricted" />
</RestrictedContainer>`

Doing that caused an undefined in AuthorizedComponent.componentWillMount() ...
`const { routes } = this.props;
`as it is no longer a "Route"

for my hard coded testing i did ...
`const routeRoles = this.props.authorize`

then I had to change 
`handleUnauthorizedRole(routeRoles, userRoles) {
        const {router} = this.context;
        router.push(this.notAuthorizedPath);
}`
to 
`handleUnauthorizedRole(routeRoles, userRoles) {
        const {router} = this.context;
        router.history.push(this.notAuthorizedPath);
}`

and of course i updated the required fields
`    static propTypes = {
        authorize: PropTypes.array.isRequired
    };`

this all works when typing the URL manually into the browser, but when accessing it via LINK then it still loads when it shouldn't.  After adding RoleAwareComponent, it loaded null but still went to my endpoint.  Is this the current behavior?

**Comment by mstruensee** [comment]
any update on this?

**Comment by jgardezi** [comment]
+1 need this.

**Comment by zenit1scada** [comment]
+1

**Comment by gaetandezeiraud** [comment]
+1

**Comment by vireshshah** [comment]
+1
